tst/EnergyPlus/unit/FaultsManager.unit.cc

```diff
@@ -148,6 +150,192 @@ TEST_F(EnergyPlusFixture, FaultsManager_FaultFoulingAirFilters_CheckFaultyAirFil
+        "  Always On Discrete,      !- Availability Schedule Name",
+        "  0.7,                     !- Fan Total Efficiency",
+        "  400,                     !- Pressure Rise {Pa}",
+        "  0.114,                   !- Maximum Flow Rate {m3/s}",
+        "  0.93,                    !- Motor Efficiency",
+        "  1,                       !- Motor In Airstream Fraction",
+        "  Node 21,                 !- Air Inlet Node Name",
+        "  Node 38;                 !- Air Outlet Node Name",
+
+        "FaultModel:Fouling:AirFilter,",
+        "  Fan CV Fouling Air Filter,  !- Name",
+        "  Fan:ConstantVolume,      !- Fan Object Type",
+        "  Fan CV,                  !- Fan Name",
+        "  AvailSched,              !- Availability Schedule Name",
+        "  Pressure Fraction Schedule,  !- Pressure Fraction Schedule Name",
+        "  Fouled Fan Curve;        !- Fan Curve Name",
+
+        "Curve:Cubic,",
+        "  Fouled Fan Curve,        !- Name",
+        "  1015,                    !- Coefficient1 Constant",
+        "  -1750,                   !- Coefficient2 x",
+        "  59050,                   !- Coefficient3 x**2",
+        "  -1624000,                !- Coefficient4 x**3",
+        "  0,                       !- Minimum Value of x",
+        "  0.09,                    !- Maximum Value of x",
+        "  ,                        !- Minimum Curve Output",
+        "  ,                        !- Maximum Curve Output",
+        "  Dimensionless,           !- Input Unit Type for X",
+        "  Dimensionless;           !- Output Unit Type",
+    });
+
+
+    // Process inputs
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    // Run CheckAndReadFaults which will call GetFanInput if not done yet
+    EXPECT_NO_THROW(CheckAndReadFaults(state));
+    compare_err_stream("", true);
+
+    DataSizing::CurZoneEqNum = 0;
+    DataSizing::CurSysNum = 0;
+    DataSizing::CurOASysNum = 0;
+
+    // DataNonZoneNonAirloopValue must be set when CurZoneEqNum and CurSysNum = 0
+    DataSizing::DataNonZoneNonAirloopValue = 0.15;
+    // We expect this one to throw, I changed the fan design pressure to 400, and made it non autosized.
+    int FanNum = 1;
+    EXPECT_ANY_THROW(Fans::SizeFan(state, FanNum));
+    EXPECT_DOUBLE_EQ(0.114, Fans::Fan(FanNum).MaxAirFlowRate);
+    std::string const error_string = delimited_string({
+        "   ** Severe  ** FaultModel:Fouling:AirFilter = \"FAN CV FOULING AIR FILTER\"",
+        "   **   ~~~   ** Invalid Fan Curve Name = \"FOULED FAN CURVE\" does not cover ",
+        "   **   ~~~   ** the operational point of Fan FAN CV",
+        "   **  Fatal  ** SizeFan: Invalid FaultModel:Fouling:AirFilter=FAN CV FOULING AIR FILTER",
+        "   ...Summary of Errors that led to program termination:",
+        "   ..... Reference severe error count=1",
+        "   ..... Last severe error=FaultModel:Fouling:AirFilter = \"FAN CV FOULING AIR FILTER\"",
+    });
```

Comment:
Added another test to show that even if fan is not autosized, it's still caught.

Reply:
The unit tests demonstrate the fix very well, although I had to add DataEnvironment::StdRhoAir = 1.2 to run it debug mode.

Reply:
:+1: